### PR TITLE
[Feature] Boolean Utilities

### LINKED ISSUE #67: [FEATURE] Boolean Utilities
## Feature Description

Introduce a set of utility functions for common boolean operations to improve code readability, reusability, and simplify logical operations in Go. Following utilities must be added. More can be added as well if person working on this has any other boolean utilities on mind
- Toggle
- AllTrue
- AnyTrue
- NoneTrue
- Equal
- CountTrue
- CountFalse
- And
- Or

---

## Use Case

- Streamlining Logical Checks:
        Example: Checking conditions for feature toggles.
- Simplifying Aggregation Logic:
        Example: Validating a series of user inputs.
- Improving Readability:
        Utility functions abstract complex logical operations into descriptive names.
- Testing and Assertions:
        Handy for automated testing where you need to count or compare boolean results.

---

## Proposed Solution

Simply add these functions in boolean package

---

## Additional Context

N/A

---

## Pseudo Code

```go
// Utilities mentioned above must follow following structure
func Toggle(value bool) bool
func AllTrue(values []bool) bool
func AnyTrue(values []bool) bool
func NoneTrue(values []bool) bool
func CountTrue(values []bool) int
func CountFalse(values []bool) int
func Equal(values ...bool) bool

// Performs operation across all elements in a slice.
func And(values []bool) bool
func Or(values []bool) bool

### PR DESCRIPTION
**Pull request by Turtel216**
### Description:
closes #67 :

Introduce a set of utility functions for common boolean operations to improve code readability, reusability, and simplify logical operations in Go.

### Checklists:
* [x] Tests passing

#### If added a new utility function or updated existing function logic:
* [x] Updated [EXAMPLES.md](/EXAMPLES.md)
* [x] Updated [readME.md](/readME.md)

## Release Notes

- **New Features**
  - Added multiple boolean utility functions: `Toggle`, `AllTrue`, `AnyTrue`, `NoneTrue`, `CountTrue`, `CountFalse`, `Equal`, `And`, and `Or`.
  - Enhanced documentation with new examples and explanations for each boolean utility function in `EXAMPLES.md` and `readME.md`.

- **Bug Fixes**
  - Improved test coverage for new boolean functions to ensure reliability and correctness.

### CODE REVIEW

File: boolean/boolean.go [hunk 1]
```diff
@@ -14,3 +14,159 @@ func IsTrue(v string) bool {
...
+/*
+CountTrue counts the number of true values in the slice.
+
+Returns:
+  - The number of elements in the slice that are true.
+*/
+func CountTrue(values []bool) int {
+	count := 0
+	for _, b := range values {
+		if b == true {
```
Comment (shahzadhaider1) [inline boolean/boolean.go:98]: just `if b` is fine

File: boolean/boolean.go [hunk 2]
```diff
@@ -14,3 +14,159 @@ func IsTrue(v string) bool {
...
+/*
+CountFalse counts the number of false values in the slice.
+
+Returns:
+  - The number of elements in the slice that are false.
+*/
+func CountFalse(values []bool) int {
+	count := 0
+	for _, b := range values {
+		if b == false {
```
Comment (shahzadhaider1) [inline boolean/boolean.go:114]: just `if !b` is fine too

**Review by shahzadhaider1 — COMMENTED** [review]
LGTM. 
couple of comments but not so serious. Will let @kashifkhan0771 review as well

**Review by kashifkhan0771 — APPROVED** [review]
LGTM!

**Comment by kashifkhan0771** [comment]
Thanks for the contribution @Turtel216 ❤️

File: boolean/boolean.go (hunk 1 above)
Comment (kashifkhan0771) [reply boolean/boolean.go:98]: Can be covered as part of tech debt ticket.

File: boolean/boolean.go (hunk 2 above)
Comment (kashifkhan0771) [reply boolean/boolean.go:114]: Can be covered as part of tech debt ticket.

**Pull request merged**